### PR TITLE
PAL-717 Added binary switch for redis master/worker or clustered

### PR DESCRIPTION
**Pull request by dbz120896**
Added a binary switch that can be used to select which Redis install to use. This does not effect the installation. That is still controlled by two other switches. This enables installation of Palisade without installing any Redis types, but selecting which existing one to use.

Could devs have a look at this. I'm sure there is something missing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,16 +117,19 @@ Some more important arguments are as follows:
 | global.kafka.install                    | Install Kafka and Zookeeper, **default=true**
 | global.redis.install                    | Install Redis, **default=true**
```
Comment (kg981167224) [inline README.md:130]: Could global.redis.install &  global.redis-cluster.install be removed now?

**Review by kg981167224 — COMMENTED** [review]
jenkinsfile has  global.redis-cluster.install references, which are currently commented out; however I think Alex will be re-enabling this very soon, so is it worth seeing if the jenkinsfile can be changed or let alex know?

File: README.md (hunk 1 above)
Comment (dbz120896) [reply README.md:130]: Those cannot be removed. The new switch is simply to select which should be used. The install flags still need to be there. It's annoying that the condition in Chart.yaml cannot support not (`!flag`). I scratched my head for a little bit about this one. I didn't want to play around with the Jenkinsfile yet.

**Review by pd104923 — DISMISSED** [review]

File: templates/NOTES.txt [hunk 2]
```diff
@@ -24,3 +24,13 @@ You have installed {{ .Chart.Name }} as release {{ .Release.Name }} into namespa
...
 {{- end }}
+
+{{- if $.Values.global.redisClusterEnabled }}
+    {{- $clusterNodes := include "palisade.redis-cluster.headlessNodes" . | trimAll "," }}
+    {{- println }}Redis is clustered using nodes: {{ $clusterNodes }}
+{{- else }}
+    {{- $masterFullname := printf "%s-master" (include "palisade.redis.fullname" .) }}
+    {{- $masterPort := int .Values.global.redis.master.service.port }}
+    {{- println }}Redis is master/worker with master at: {{ $masterFullname }}:{{ $masterPort }}
+{{- end }}
```
Comment (dev930018) [inline templates/NOTES.txt:35]: I'd like to see a warning here for the specific case when redis(cluster) is set to install but does not match the flag:
```
if (redisClusterEnabled and redis.install) or (not redisClusterEnabled and redisCluster.install)
println Redis install type does not match redis deployment type, are you sure you know what you are doing?
```

**Review by dev930018 — CHANGES_REQUESTED** [review]

File: templates/NOTES.txt (hunk 2 above)
Comment (dbz120896) [reply templates/NOTES.txt:35]: I found that you can now get it to fail if you have an install/use mismatch, which I think is better. If no installation is selected, then no error will be thrown.

**Review by dev930018 — DISMISSED** [review]

**Review by ac74475 — CHANGES_REQUESTED** [review]
Given the limitations on the charts dependancy conditionals, I think the best option is to work with just the original two Redis install parameters and have a check that will fail the deployment if both are set too true. Rather than adding in another parameter.

**Review by ac74475 — APPROVED** [review]

**Pull request merged**